### PR TITLE
Fix amd64 floating point control functions

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

- Add test for 
- Implement amd64 versions of _controlfp, _control87, _clearfp, _fpreset, _statusfp
- Implement amd64 KiXmmExceptionHandler

## TODO

- [ ] Fix test for x86

### CODE REVIEW

**Comment by julenuri** [comment]
About CORE-18102, the retest make me to reach a message that closes as soon as the software starts.
VirtualBox_ReactOS x64_15_07_2022_12_40_50
But it's not freezing the system as the ticket report anymore.

**Comment by julenuri** [comment]
After last commits: :) Peazip
VirtualBox_ReactOS x64_16_07_2022_18_06_34

File: ntoskrnl/ke/amd64/except.c [hunk 1]
```diff
@@ -660,7 +660,48 @@ NTAPI
...
+    }
+    else if ((TrapFrame->MxCsr & _MM_EXCEPT_INEXACT) &&
+             !(TrapFrame->MxCsr & _MM_MASK_INEXACT))
+    {
+        /* Precision */
+        ExceptionCode = STATUS_FLOAT_INEXACT_RESULT;
+    }
+    else
+    {
+        ASSERT(FALSE);
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply ntoskrnl/ke/amd64/except.c:703]: Should never happen.

File: ntoskrnl/ke/amd64/cpu.c [hunk 2]
```diff
@@ -375,7 +375,7 @@ KiRestoreProcessorControlState(PKPROCESSOR_STATE ProcessorState)
 //    __ltr(&ProcessorState->SpecialRegisters.Tr);
     __lidt(&ProcessorState->SpecialRegisters.Idtr.Limit);
 
-//    __ldmxcsr(&ProcessorState->SpecialRegisters.MxCsr); // FIXME
+    _mm_setcsr(ProcessorState->SpecialRegisters.MxCsr); // FIXME
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply ntoskrnl/ke/amd64/cpu.c]: nothing, removed it

**Pull request merged**